stdlib/REPL/test/replcompletions.jl

```diff
@@ -1145,6 +1145,14 @@ let s = "test_dict[\"ab"
     @test c == Any["\"abc\"", "\"abcd\""]
 end
 
+primitive type NonStruct 8 end
+Base.propertynames(::NonStruct) = (:a, :b, :c)
+x = reinterpret(NonStruct, 0x00)
```

Comment:
Move this into `module CompletionFoo` at the top of the file. The REPL runs in module Main, while here the test runs in an anonymous module.

(and make the test for `s = "CompletionFoo.x."`)